platform-docs/docs/intro.md

```diff
@@ -10,76 +10,105 @@ Let's discover how to use the **Gutenberg Block Editor** to build your own block
+We're going to be using JSX to write our UI and components as the block editor is built with React. Using the Vite bootstrap described above there’s no need to configure anything as it outputs the result as a React pragma. If you're using a different bundler/build tool, you may need to configure the JSX transpilation to do the same.
 
 ## Bootstrap your block editor
 
-It's time to render our first block editor.
+It's time to render our first block editor. Update your `src/App.jsx` file with the following code:
 
- - Update your `index.jsx` file with the following code:
 ```jsx
-import { createElement, useState } from "react";
-import { createRoot } from 'react-dom/client';
+import { useState } from "react";
 import {
   BlockEditorProvider,
   BlockCanvas,
 } from "@wordpress/block-editor";
 import { registerCoreBlocks } from "@wordpress/block-library";
+import { getBlockTypes } from "@wordpress/blocks";
 
 // Default styles that are needed for the editor.
-import "@wordpress/components/build-style/style.css";
-import "@wordpress/block-editor/build-style/style.css";
+import componentsStyles from "@wordpress/components/build-style/style.css?raw";
+import blockEditorStyles from "@wordpress/block-editor/build-style/style.css?raw";
+import blockEditorContentStyles from "@wordpress/block-editor/build-style/content.css?raw";
 
 // Default styles that are needed for the core blocks.
-import "@wordpress/block-library/build-style/common.css";
-import "@wordpress/block-library/build-style/style.css";
-import "@wordpress/block-library/build-style/editor.css";
-
-// Register the default core block types.
-registerCoreBlocks();
-
-function Editor() {
+import blocksCommonStyles from "@wordpress/block-library/build-style/common.css?raw";
+import blocksStyles from "@wordpress/block-library/build-style/style.css?raw";
+import blocksEditorStyles from "@wordpress/block-library/build-style/editor.css?raw";
+
+const styles = `
+  ${ componentsStyles }
+  ${ blockEditorStyles }
+  ${ blockEditorContentStyles }
+  ${ blocksCommonStyles }
+  ${ blocksStyles }
+  ${ blocksEditorStyles }
+`;
+
+const registeredBlockTypes = getBlockTypes();
+// Registers the default core block types, avoiding doing so again during HMR.
+if (
+  ! registeredBlockTypes.length ||
+  ! registeredBlockTypes.some((blockType) => blockType.name.startsWith('core/'))
+) registerCoreBlocks();
+
+export default function Editor() {
   const [blocks, setBlocks] = useState([]);
   return (
-    /*
-      The BlockEditorProvider is the wrapper of the block editor's state.
-      All the UI elements of the block editor need to be rendered within this provider.
-    */
-    <BlockEditorProvider
-      value={blocks}
-      onChange={setBlocks}
-      onInput={setBlocks}
-    >
+    <>
+      <style>{ styles }</style>
```

Comment:
Let's avoid this and just do `import`for the two stylesheets that need to load here. We should clarify what things are "content" styles and what things are "UI styles". I think we should add a paragraph to explain this.

Reply:
Okay. This `style` tag is gone. This reason it was this way was because I’d added the `?raw` param to the their import path to tell Vite not to process them and due to that Vite won’t automatically inject them. I can’t remember if there was any issue with Vite processing them or if I just thought it was better dev performance. Now instead of doing an `import` for those files I’ve instructed that they be added to `index.html` (the `vite-ignore` attribute will also keep them from being processed).

I know it’s not quite as simple that way but I think it may help express the difference with these UI styles versus the content styles. I also added a note hoping to help explain it, not sure if it suffices.